### ISSUE TITLE
quotes rule confused by `""''`

### ISSUE THREAD

**Issue by wincent** [eslint/eslint#9433]
**Tell us about your environment**

* **ESLint Version:** 4.8.0
* **Node Version:** 8.6.0
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:** 

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "node": true
  },
  "parser": "babel-eslint",
  "rules": {
    // key: 0 = allow, 1 = warn, 2 = error
    "comma-dangle": [0, "always-multiline"],
    "global-strict": [0],
    "no-process-exit": [0],
    "no-underscore-dangle": [0],
    "no-use-before-define": [0],
    "quotes": [1, "single", "avoid-escape"],
    "strict": [2, "never"]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const md = MarkdownIt({
  quotes: `""''`,
});
```
**What did you expect to happen?**

No lints.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/glh/code/masochist/src/server/schema/types/MarkupType.js
  68:11  warning  Strings must use singlequote  quotes

✖ 1 problem (0 errors, 1 warning)
  0 errors, 1 warning potentially fixable with the `--fix` option.
```

**Comment by kaicataldo** [comment]
I've reproduced this locally with the following minimal repro case using the default parser:

Config:
```json
{
  "parserOptions": {
        "ecmaVersion": 6,
  },
  "rules": {
    "quotes": [1, "single", "avoid-escape"]
  }
}
```
```js
var foo = `""''`;
```
```sh
  1:11  warning  Strings must use singlequote  quotes

✖ 1 problem (0 errors, 1 warning)
  0 errors, 1 warning potentially fixable with the `--fix` option.
```

**Comment by not-an-aardvark** [comment]
Related: https://github.com/eslint/eslint/issues/7660

**Comment by kaicataldo** [comment]
I thought I'd seen this before!

**Comment by kaicataldo** [comment]
From #7660:
> This would change the default behavior of the avoidEscape option. Originally I thought the existing behavior was a bug, but based on the option description it appears to be working as documented.

As @not-an-aardvark points out, it does indeed look like this is the intended behavior and this would be an enhancement to the rule.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
